### PR TITLE
Issue #14: Import iOS editor

### LINKED ISSUE #14: Main editor functionality
Import the iOS editor and successfully load content into the JS-based editor.

### PR DESCRIPTION
**Pull request by aforcier**
Fixes #14. Updated the JavaScript editor code from the iOS project and set up auto-loading an example HTML file.
- Turned on web debugging
- Changed the `WordPress-Utils-Android` dependency to use the Maven artifact
- Added build instructions to the `README` for Windows (fixes a Windows-only missing symlink issue)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
+    mklink /D [PROJECT_ROOT]\WordPressEditor\src\main\assets %PROJECT_ROOT%\libs\editor-common\assets
+
+Finally, update `[PROJECT_ROOT]\.git\info\exclude` to ignore the symlink locally:
+
+    # assets symlink
+    WordPressEditor/src/main/assets
+
+## LICENSE ##
+
+This library is licensed under [MIT](LICENSE-MIT).
```
Comment (bummytime) [inline README.md]: Let's keep this consistent with wpandoid...[GPLv2](https://github.com/wordpress-mobile/WordPress-Android#license)

**Comment by bummytime** [comment]
@aforcier I tested the example app on a Galaxy S2, 4.1.1 (API 16) and was not able to load the test post. This error popped up a lot:

```
04-03 16:25:13.780    2028-2028/org.wordpress.editorexample E/WordPress-EDITOR﹕ The protocol isn't supported.
```

screen shot 2015-04-03 at 11 21 10 am

Higher OS/API levels did work.

File: WordPressEditor/src/main/java/org/wordpress/android/editor/EditorFragment.java [hunk 2]
```diff
@@ -163,4 +178,32 @@ public void appendGallery(MediaGallery mediaGallery) {
...
+                mWebView.post(new Runnable() {
+                    public void run() {
+                        String htmlFile = "";
+                        try {
+                            htmlFile = getStringFromAsset("example-content.html");
+                            htmlFile = htmlFile.replace("\\", "\\\\");
+                            htmlFile = htmlFile.replace("\"", "\\\"");
+                            htmlFile = htmlFile.replace("'", "\\'");
+                            htmlFile = htmlFile.replace("\r", "\\r");
+                            htmlFile = htmlFile.replace("\n", "\\n");
```
Comment (bummytime) [inline WordPressEditor/src/main/java/org/wordpress/android/editor/EditorFragment.java]: Could we refactor a lot of this into a stand-alone utility method?

File: WordPressEditor/src/main/java/org/wordpress/android/editor/EditorFragment.java [hunk 3]
```diff
@@ -81,23 +83,28 @@ public void onReceivedError(WebView view, int errorCode, String description, Str
...
 
             @Override
             public void onConsoleMessage(String message, int lineNumber, String sourceId) {
-                AppLog.e(T.EDITOR, message + " -- from line " + lineNumber + " of " + sourceId);
+                AppLog.d(T.EDITOR, message + " -- from line " + lineNumber + " of " + sourceId);
             }
         });
         String htmlEditor = getHtmlEditor();
+
+        mWebView.addJavascriptInterface(new JsCallbackHandler(), "nativeCallbackHandler");
```
Comment (bummytime) [inline WordPressEditor/src/main/java/org/wordpress/android/editor/EditorFragment.java]: Should `nativeCallbackHandler` be a const?

**Comment by bummytime** [comment]
Alright @aforcier... just a few things. Let me know your thoughts. Nice job!

**Comment by aforcier** [comment]
Updated to address @bummytime's comments:
- License changed to GPLv2
- Fixed the test post not loading on earlier API versions. Lesson: use `javascript` instead of `JavaScript` when calling JS using `WebView.loadUrl`, otherwise it breaks JS on `API<19`
- Some refactoring of EditorFragment. Opened an issue for some further refinement here: #23

**Comment by bummytime** [comment]
Awesome sauce @aforcier! :shipit:

**Pull request merged**